src/Form/index.tsx

```diff
@@ -1748,12 +1750,44 @@ function Form({
           setElementError('Your phone number is invalid or requested too much');
           break;
         }
+      } else if (type === ACTION_SEND_EMAIL_CODE) {
+        const emailAddress = fieldValues[action.email_field_key] as string;
+        if (validators.email(emailAddress)) {
+          try {
+            await client.sendEmailOTP(emailAddress);
+          } catch (e) {
+            setElementError((e as Error).message);
+            break;
+          }
+        } else {
+          setElementError('Your target email address is invalid or requested too much');
+          break;
+        }
       } else if (type === ACTION_VERIFY_SMS) {
         const pinKey = action.auth_target_field_key;
         const pin = fieldValues[pinKey] as string;
         const params = { fieldVal: pin, featheryClient: client };
         let hasErr = false;
-        await Auth.verifySms(params).catch((e: Error) => {
+        await Auth.verifySMSOTP(params).catch((e: Error) => {
+          let message = '';
+          if (e.message === 'Please try again')
+            message = 'Your code is invalid';
+          else if (e.message in initState.defaultErrors)
+            message = initState.defaultErrors[e.message];
+          else message = e.message;
+          setElementError(message);
+          hasErr = true;
+        });
+        if (hasErr) break;
+
+        client.submitCustom({ [pinKey]: pin });
+        authState.redirectAfterLogin = true;
+      } else if (type === ACTION_VERIFY_EMAIL) {
```

Comment:
since the verify email and verify sms actions are the same, can we put them in the same if statement?

else if ([ACTION_VERIFY_EMAIL, ACTION_VERIFY_SMS].includes(type)) {
}